src/components/Header.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Types';
+import { showError } from '../helpers/helpers';
+import { updateTodo, addTodos, USER_ID } from '../api/todos';
+
+interface Props {
+  todos: Todo[],
+  setError: React.Dispatch<React.SetStateAction<string>>;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setLoader: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setError,
+  setTempTodo,
+  setTodos,
+  setLoader,
+}) => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [submitDisable, setSubmitDisable] = useState(false);
+  const activeInputButton = todos.every(todo => todo.completed);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    (setSearchQuery(event.target.value));
+  };
+
+  const newTodo = {
+    id: 0,
+    title: searchQuery,
+    userId: USER_ID,
+    completed: false,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setSubmitDisable(true);
+    if (!searchQuery) {
+      showError('Title can\'t be empty', setError);
+      setSubmitDisable(false);
+    } else {
+      setLoader(prevState => [...prevState, newTodo.id]);
+      addTodos(USER_ID, searchQuery)
+        .then(todo => {
+          setTodos(prevState => {
+            setTempTodo(null);
+            setSubmitDisable(false);
+            setSearchQuery('');
+
+            return [...prevState, todo];
+          });
+        })
+        .catch(() => showError('Unable to add a todo', setError));
+      setTempTodo(newTodo);
+    }
+  };
+
+  const makeAllTodosActive = () => {
+    todos.forEach(todo => {
+      updateTodo(todo.id, { completed: !todo.completed })
+        .then((updatedTodo) => {
+          setTodos(prevState => (
+            prevState.map(prevTodo => {
+              return (prevTodo.id === todo.id) ? updatedTodo : prevTodo;
```

Comment:
ternaries should be multilined, as it is super difficult to read if written on a single